src/workerd/api/node/url.c++

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2017-2022 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+#include "url.h"
+
+#include <ada.h>
+
+namespace workerd::api::node {
+
+kj::ArrayPtr<const char> UrlUtil::domainToUnicode(kj::String domain) {
+  if (domain.size() == 0) {
+    return {};
+  }
+
+  auto out = ada::parse<ada::url_aggregator>("ws://x");
+  JSG_REQUIRE(out.has_value(), Error, "URL parsing failed"_kj);
+  if (!out->set_hostname({domain.begin(), domain.size()})) {
+    return {};
+  }
+  auto result = out->get_hostname();
+  return kj::ArrayPtr<const char>(result.data(), result.size());
+}
+
+kj::ArrayPtr<const char> UrlUtil::domainToASCII(kj::String domain) {
+  if (domain.size() == 0) {
+    return {};
+  }
+
+  auto out = ada::parse<ada::url_aggregator>("ws://x");
```

Comment:
I'm guessing `ws://x` is a fake URL to initialize with? If so, please make a constant for it.

Reply:
Making a constant means that the URL is shared between multiple calls. Technically that would be faster, but for the sake of being 100% compliant with Node.js, I prefer to keep it as it is.